Dev environment fix

Slaith12 opened the pull request:
# Feature
Makes various fixes to allow the program to run on our local (windows) computers.
Notable changes:
Logger uses a different filename when run on dev environment
**!!!** .env file is removed from .gitignore, and added to repo (not sure if this causes problems when pushed to the pi, please review)
Added additional check for is_dev() to detect missing .env file.

# How to Test
Run the cli command "catbot start -D" on your local machine. The GUI should appear. It will crash if you press certain keys, but the main thing this pr is changing is the setup to make sure it works on our computers.

Slaith12 committed:
Fix log file name and .env file

Windows does not like colons in the filename. It also gets confused when you copy the **entire file path** into the file name.

As for .env, the file just wasn't in my repo and the is_dev function didn't catch it. I changed the .gitignore to fix this, but we should address why it was in the gitignore in the first place

Slaith12 committed:
Program no longer immediately crashes

The program will still crash if you press "up" on your keyboard, but you at least get to the control ui

yes this is a shifty way to fix the crash, but I can not find where pull_up is even added to the call context

hiromon0125 commented on .gitignore:
```diff
@@ -2,7 +2,8 @@
 .vscode/arduino.json
-**/.env
+# we need the .env file in order for is_dev to work correctly
+# **/.env
```
This is definitely not true. You have .env-example that can be copied.

Slaith12 replied:
You should definitely mention somewhere (either in the readme or the wiki) that we need to copy that file in order for the program to work, since new people would assume stuff like this is already handled otherwise.

hiromon0125 replied:
Ah I see you can update the readme file for that then.

hiromon0125 commented on src/raspi/.env:
Don't rename this

hiromon0125 commented on src/raspi/state_management/utils/deviceMock.py:
```diff
@@ -58,10 +58,11 @@ class FakeDigitalInputDevice:
     when_deactivated = None
 
-    def __init__(self, pin: int, initial_value: int = 0, initial_is_state=False):
+    def __init__(self, pin: int, initial_value: int = 0, initial_is_state=False, pull_up = False):
```
pull_up is a way to keep the input pin in normally open and keeping it open.

hiromon0125 replied:
One of the component uses this parameter so definitely a good catch

hiromon0125 commented on src/raspi/state_management/utils/logger.py:
```diff
@@ -30,11 +32,22 @@ def configure_logger(level: str = "Debug"):
+            format="%(filename)s: %(message)s",
+            level=lvl,  # TODO: hook it to env or config file
+            force=True,
+        )
```
What is the benefit of this?

Slaith12 replied:
The normal logger file name causes a crash on our local machine because it uses invalid characters (colons). Also, including the "filename" variable in the file name causes the entire file path to be copied into the name, including /'s. I don't know if there's any importance for the loggers to use the old file name when running on the pi, so I kept that there just in case.

hiromon0125 replied:
oh I did not realize this was a thing lol. You can make this a default for both so its the same convention on every device.

hiromon0125 reviewed (changes requested):
check my comments

Slaith12 committed:
Revert .env changes

Slaith12 committed:
Standardize logger filename

The file name no longer contains the program's name, which can be annoying if you're running multiple different samples and need the loggers, but Hiro said it'd be fine to leave it like this
